src/ApparentHorizons/StrahlkorperInDifferentFrame.cpp

```diff
@@ -0,0 +1,268 @@
+  // so, for each index 's' (corresponding to an angular collocation
+  // point), find the root 'r_dest' that zeroes this lambda function.
+  //
+  // This version of the function returns a std::optional<double>
+  // because it might fail if r_dest is out of range of the map. Below
+  // there is another version of the function that returns a double.
+  const auto radius_function_for_bracketing =
+      [&r_hat, &center_dest, &src_strahlkorper, &domain, &functions_of_time,
+       &time](const double r_dest,
+              const size_t s) noexcept -> std::optional<double> {
+    // Get destination Cartesian coordinates of the point.
+    const tnsr::I<double, 3, DestFrame> x_dest{
+        {r_dest * get<0>(r_hat)[s] + center_dest[0],
+         r_dest * get<1>(r_hat)[s] + center_dest[1],
+         r_dest * get<2>(r_hat)[s] + center_dest[2]}};
+
+    // Transform to source Cartesian coordinates of the point.  This
+    // must be done block by block, since only one block will have the
+    // source coordinates (except on block boundaries, in which case
+    // multiple blocks will succeed; we use the first block that succeeds
+    // since they all should give the same result modulo roundoff).
+    for (const auto& block : domain.blocks()) {
+      // Once there are more possible source frames than the grid frame
+      // (i.e. the distorted frame), then this static_assert will change,
+      // and there will be an `if constexpr` below to treat the different
+      // possible source frames.
+      static_assert(std::is_same_v<SrcFrame, ::Frame::Grid>,
+                    "Source frame must currently be Grid frame");
+      static_assert(std::is_same_v<DestFrame, ::Frame::Inertial>,
+                    "Destination frame must currently be Inertial frame");
+      const auto x_src = block.moving_mesh_grid_to_inertial_map().inverse(
+          x_dest, time, functions_of_time);
+      if (x_src.has_value()) {
+        // Sometimes the inverse map might be defined, but the logical
+        // coordinates might be outside the correct range. So go back
+        // to logical coords and make sure that the logical coords are
+        // between -1 and 1.
+        const auto x_logical =
+            block.moving_mesh_logical_to_grid_map().inverse(x_src.value());
+        if (x_logical.has_value() and get<0>(x_logical.value()) <= 1.0 and
+            get<0>(x_logical.value()) >= -1.0 and
+            get<1>(x_logical.value()) <= 1.0 and
+            get<1>(x_logical.value()) >= -1.0 and
+            get<2>(x_logical.value()) <= 1.0 and
+            get<2>(x_logical.value()) >= -1.0) {
+          // Find (r_src,theta_src,phi_src) in source coordinates.
+          const double r_src = sqrt(square(get<0>(x_src.value())) +
+                                    square(get<1>(x_src.value())) +
+                                    square(get<2>(x_src.value())));
+          const double theta_src = acos(get<2>(x_src.value()) / r_src);
```

Comment:
`atan2(sqrt(x^2 + y^2), z)` (You might also like `std::hypot` to reduce typing, but that's optional.)